test/e2e/framework/framework.go

```diff
@@ -350,6 +354,71 @@ func nextAzureVMSize(current string) (string, error) {
 	return fmt.Sprintf("Standard_%s%d%s%s", family, multiplier, subfamily, version), nil
 }
 
+// increaseGCPInstanceSize increases the instance size of the instance on the providerSpec for an GCP providerSpec.
+func increaseGCPInstanceSize(rawProviderSpec *runtime.RawExtension, providerConfig providerconfig.ProviderConfig) error {
+	cfg := providerConfig.GCP().Config()
+
+	var err error
+
+	cfg.MachineType, err = nextGCPMachineSize(cfg.MachineType)
+	if err != nil {
+		return fmt.Errorf("failed to get next instance size: %w", err)
+	}
+
+	if err := setProviderSpecValue(rawProviderSpec, cfg); err != nil {
+		return fmt.Errorf("failed to set provider spec value: %w", err)
+	}
+
+	return nil
+}
+
+// nextGCPVMSize returns the next GCP machine size in the series.
+// The Machine sizes being used are in format <e2|n2|n1>-standard-<number>.
+func nextGCPMachineSize(current string) (string, error) {
+	// Regex to match the GCP machine size string.
+	re := regexp.MustCompile(`(?P<family>[0-9a-z]+)-standard-(?P<version>[0-9]+)`)
```

Comment:
I think this should be multiplier right?
```suggestion
	re := regexp.MustCompile(`(?P<family>[0-9a-z]+)-standard-(?P<multiplier>[0-9]+)`)
```

Reply:
Makes sense